src/main/java/module/transfer/scout/Player.java

```diff
@@ -557,30 +545,37 @@ public final boolean isHomwGrown() {
-        buffer.append(", price = " + price);
-        buffer.append(", playerId = " + playerId);
-        buffer.append(", speciality = " + speciality);
-        buffer.append(", TSI = " + tsi);
-        buffer.append(", Age = " + age +"."+ageDays);
-        buffer.append(", expiryDate = " + expiryDate + " " + expiryTime);
-        buffer.append(", experience = " + experience);
-        buffer.append(", form = " + form);
-        buffer.append(", stamina = " + stamina);
-        buffer.append(", goalKeeping = " + goalKeeping);
-        buffer.append(", playMaking = " + playMaking);
-        buffer.append(", passing = " + passing);
-        buffer.append(", wing = " + wing);
-        buffer.append(", defense = " + defense);
-        buffer.append(", attack = " + attack);
-        buffer.append(", setPieces = " + setPieces);
-        buffer.append(", loyalty = " + loyalty);
-        buffer.append(", homegrown = " + homegrown);
-        buffer.append(", agreeability = " + agreeability);
-        buffer.append(", baseWage = " + baseWage);
-        buffer.append(", nationality = " + nationality);
-        buffer.append(", info = " + getInfo());
+        final StringBuilder buffer = new StringBuilder();
+        buffer.append("playerName = ").append(playerName);
+        buffer.append(", price = ").append(price);
+        buffer.append(", playerId = ").append(playerId);
+        buffer.append(", speciality = ").append(speciality);
+        buffer.append(", TSI = ").append(tsi);
+        buffer.append(", Age = ").append(age).append(".").append(ageDays);
+        buffer.append(", expiryDate = ");
+        if ( expiryDate!= null) buffer.append(expiryDate.toLocaleDateTime());
+        buffer.append(", experience = ").append(experience);
+        buffer.append(", form = ").append(form);
+        buffer.append(", stamina = ").append(stamina);
+        buffer.append(", goalKeeping = ").append(goalKeeping);
+        buffer.append(", playMaking = ").append(playMaking);
+        buffer.append(", passing = ").append(passing);
+        buffer.append(", wing = ").append(wing);
+        buffer.append(", defense = ").append(defense);
+        buffer.append(", attack = ").append(attack);
+        buffer.append(", setPieces = ").append(setPieces);
+        buffer.append(", loyalty = ").append(loyalty);
+        buffer.append(", homegrown = ").append(homegrown);
+        buffer.append(", agreeability = ").append(agreeability);
+        buffer.append(", baseWage = ").append(baseWage);
+        buffer.append(", nationality = ").append(nationality);
+        buffer.append(", info = ").append(getInfo());
         return buffer.toString();
     }
+
+    public void setExpiryDate(String dateString, String timeString) {
+        var datetimeformatter = DateTimeFormatter.ofLocalizedDateTime(FormatStyle.MEDIUM, FormatStyle.SHORT);
+        var localDateTime = LocalDateTime.parse(dateString+", "+ timeString, datetimeformatter);
```

Comment:
german format gives an exception if that comma is missing - how to determine the correct formatter???

Reply:
I have been scratching my head over that one too, because the format above `02-05-2023 20.11` also fails with the locale I have on my machine because time would be `20:11` and I am not sure about date separator either.

I had checked whether that format was available from CHPP, because couldn't see it.

Reply:
Next attempt to convert the date

Reply:
I still don't get how the format for the deadline date is selected in Hattrick. I have changed the language in the Preferences to English (US), French, German, Arabic, and I still get the same DD-MM-YYYY HH.MM format.

I have also changed Firefox settings to French, German, Arabic, and again, same format... Without knowing what is uses, it's somewhat hard to figure out how to parse this correctly.

Reply:
@wsbrenk Do you mind if I push a commit to this PR?  I have a solution that uses `getDeadlineString`.

Reply:
@tychobrailleur no, it is fine if you commit to this PR.

Reply:
I have just pushed.